controllers/user.controller.js

```diff
@@ -1,25 +1,90 @@
-const httpStatus = require("http-status");
-const User = require("../models/user.model");
-const sendResponse = require("../helpers/response");
+const httpStatus = require('http-status');
+const User = require('../models/user.model');
+const sendResponse = require('../helpers/response');
+const { createUser } = require('../validations/user.validation');
+const gravatar = require('gravatar');
+const { Joi } = require('celebrate');
+const pick = require('ramda/src/pick');
 
 /**
  * Load user and append to req.
  */
 exports.load = async (req, res, next, id) => {
   try {
     // Load user object from quarystring Id
-    return res.json(sendResponse(
-      httpStatus.NOT_FOUND,
-      "No such user exists!",
-      null,
-      null
-    ))
+    return res.json(
+      sendResponse(httpStatus.NOT_FOUND, 'No such user exists!', null, null)
+    );
   } catch (error) {
     next(error);
   }
-}; 
+};
 
 exports.getUsers = (req, res) => {
-  return res.json(sendResponse(200,'testing', null, null))
-}
+  return res.json(sendResponse(200, 'testing', null, null));
+};
 
+exports.signup = async (req, res) => {
+  const { error } = Joi.validate(req.body, createUser.body);
+  if (error)
+    return res.json(
+      sendResponse(
+        httpStatus.BAD_REQUEST,
+        'Bad Request',
+        null,
+        error.details[0].message
+      )
+    );
+
+  try {
+    const { name, email, password, isMentor, isAdmin } = req.body;
+
+    //check if user exists
+    let user = await User.getByEmail(email);
+    if (user) {
+      return res.json(
+        sendResponse(httpStatus.BAD_REQUEST, 'Bad Request', null, {
+          msg: 'Email already in use!'
+        })
+      );
+    }
+
+    //get user avatar
+    const avatar = gravatar.url(email, {
```

Comment:
This can be used as a default value in the mongoose schema, 
```javascript
// user.model.js
avatar : {
    type: String,
    default : function(v = this.email) { return getAvatar(v))
}
// then in the helper, you could do something like this
// helper/avatar.js
module.exports = (email) => {
    gravatar.url(email, {
    // ...
})
}
```